naga/src/back/msl/writer.rs

```diff
@@ -1150,14 +1151,73 @@ impl<W: Write> Writer<W> {
             write!(self.out, " ? ")?;
         }
 
-        write!(
-            self.out,
-            "{NAMESPACE}::atomic_{key}_explicit({ATOMIC_REFERENCE}"
-        )?;
-        self.put_access_chain(pointer, policy, context)?;
-        write!(self.out, ", ")?;
-        self.put_expression(value, context, true)?;
-        write!(self.out, ", {NAMESPACE}::memory_order_relaxed)")?;
+        fn get_msl_non_weak_exchange_name(
```

Comment:
There's no reason this needs to be inlined here, where the details are a distraction.

#5383 already moves the function out of `msl/mod.rs`. I think this PR should leave it where it landed.